service/arm_templates/network_setup.json

```diff
@@ -0,0 +1,221 @@
+
+    "mastersSecurityGroupName": "[concat(uniquestring(resourceGroup().id), '-MastersSecurityGroup')]",
+    "mastersSecurityGroupID": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('mastersSecurityGroupName'))]",
+    "mastersSubnetName": "[concat(variables('virtualNetworkName'), '-MastersSubnet')]",
+    "mastersSubnetID": "[concat(variables('virtualNetworkID'), '/subnets/', variables('mastersSubnetName'))]",
+
+    "workersSecurityGroupName": "[concat(uniquestring(resourceGroup().id), '-WorkersSecurityGroup')]",
+    "workersSecurityGroupID": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('workersSecurityGroupName'))]",
+    "workersSubnetName": "[concat(variables('virtualNetworkName'), '-WorkersSubnet')]",
+    "workersSubnetID": "[concat(variables('virtualNetworkID'), '/subnets/', variables('workersSubnetName'))]"
+
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Network/virtualNetworks",
+      "name": "[variables('virtualNetworkName')]",
+      "apiVersion": "[parameters('virtualNetworksAPIVersion')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/networkSecurityGroups/', variables('mastersSecurityGroupName'))]",
+        "[concat('Microsoft.Network/networkSecurityGroups/', variables('workersSecurityGroupName'))]"
+      ],
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[parameters('addressPrefix')]"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "[variables('mastersSubnetName')]",
+            "properties": {
+              "addressPrefix": "[parameters('subnetMasterCIDR')]",
+              "networkSecurityGroup": {
+                "id": "[variables('mastersSecurityGroupID')]"
+              }
+            }
+          },
+          {
+            "name": "[variables('workersSubnetName')]",
+            "properties": {
+              "addressPrefix": "[parameters('subnetWorkerCIDR')]",
+              "networkSecurityGroup": {
+                "id": "[variables('workersSecurityGroupID')]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "name": "[variables('mastersSecurityGroupName')]",
+      "apiVersion": "[parameters('networkSecurityGroupsAPIVersion')]",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "securityRules": [
+          {
+            "name": "defaultInboundRule",
+            "properties": {
+              "description": "Default rule that denies any inbound traffic.",
+              "protocol": "*",
+              "sourcePortRange": "*",
+              "destinationPortRange": "*",
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "[parameters('subnetMasterCIDR')]",
+              "access": "Deny",
+              "direction": "Inbound",
+              "priority": 1000
```

Comment:
Do all security group rules need to have a priority? Is there any benefit in listing the rules by priority in the template?